packages/atomic/src/components/facets-v1/atomic-category-facet/atomic-category-facet.tsx

```diff
@@ -0,0 +1,313 @@
+   */
+  @Prop() public withSearch = false;
+  /**
+   * The sort criterion to apply to the returned facet values.
+   * Possible values are 'score', 'alphanumeric', 'occurrences', and 'automatic'.
+   * TODO: add automatic (occurences when not expanded, alphanumeric when expanded)
+   */
+  @Prop() public sortCriteria: CategoryFacetSortCriterion = 'occurrences';
+  /**
+   * The character that separates values of a multi-value field.
+   */
+  @Prop() public delimitingCharacter = ';';
+  /**
+   * The base path shared by all values for the facet, separated by commas.
+   */
+  @Prop() public basePath?: string;
+  /**
+   * Whether to use basePath as a filter for the results.
+   */
+  @Prop() public filterByBasePath = true;
+  // @Prop() public customSort?: string; TODO: KIT-753 add customSort to headless
+
+  public initialize() {
+    this.searchStatus = buildSearchStatus(this.bindings.engine);
+    const options: CategoryFacetOptions = {
+      facetId: this.facetId,
+      field: this.field,
+      numberOfValues: this.numberOfValues,
+      sortCriteria: this.sortCriteria,
+      facetSearch: {numberOfValues: this.numberOfValues},
+      basePath: this.basePath
+        ? this.basePath.split(',').map((value) => value.trim())
+        : undefined,
+      delimitingCharacter: this.delimitingCharacter,
+      filterByBasePath: this.filterByBasePath,
+    };
+    this.facet = buildCategoryFacet(this.bindings.engine, {options});
+    this.facetId = this.facet.state.facetId;
+    this.bindings.store.state.facets[this.facetId] = {
+      label: this.label,
+    };
+  }
+
+  public componentShouldUpdate(
+    next: unknown,
+    prev: unknown,
+    propName: keyof AtomicCategoryFacet
+  ) {
+    if (propName === 'facetState') {
+      return shouldUpdateFacetSearchComponent(
+        (next as CategoryFacetState).facetSearch,
+        (prev as CategoryFacetState).facetSearch
+      );
+    }
+
+    return true;
+  }
+
+  private get numberOfSelectedValues() {
+    return this.facetState.values.filter(({state}) => state === 'selected')
+      .length;
+  }
+
+  private renderHeader() {
+    return (
+      <FacetHeader
+        i18n={this.bindings.i18n}
+        label={this.label}
+        onClearFilters={() => this.facet.deselectAll()}
+        numberOfSelectedValues={this.numberOfSelectedValues}
+        isCollapsed={this.isCollapsed}
+        onToggleCollapse={() => (this.isCollapsed = !this.isCollapsed)}
+      ></FacetHeader>
+    );
+  }
+
+  private renderSearchInput() {
+    if (!this.withSearch) {
+      return;
+    }
+
+    return (
+      <FacetSearchInput
+        i18n={this.bindings.i18n}
+        label={this.label}
+        query={this.facetState.facetSearch.query}
+        onChange={(value) => {
+          if (value === '') {
+            this.facet.facetSearch.clear();
+            return;
+          }
+          this.facet.facetSearch.updateCaptions(
+            getFieldCaptions(this.field, this.bindings.i18n)
+          );
```

Comment:
Should this be in `initialize`?

Reply:
We dont have access to the strings on initialize 

We fetch and update the state so that if the strings get updated by the customer, or the language changes, we get the update